### PR TITLE
Fix column mapping in output_ordering() and output_partitioning() for ProjectionExec and AggregateExec

### LINKED ISSUE #6099: wrong result when distinct aggregate.
### Describe the bug

```sql
CREATE TABLE tab0(col0 INTEGER, col1 INTEGER, col2 INTEGER)
INSERT INTO tab0 VALUES(83,0,38)
INSERT INTO tab0 VALUES(26,0,79)
INSERT INTO tab0 VALUES(43,81,24)

SELECT DISTINCT 94 AS col1 FROM tab0 AS cor0 GROUP BY cor0.col1
----
94
94

```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### LINKED ISSUE #6112: output_ordering() and output_partitioning() in ProjectionExec is wrong
### Describe the bug

Can not derive the output ordering and output partition from input directly.

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mingmwang**
# Which issue does this PR close?

Closes #6099
Closes #6112

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/groupby.slt [hunk 1]
```diff
@@ -1477,14 +1477,11 @@ SELECT + cor0.col0 AS col2 FROM tab1 AS cor0 GROUP BY cor0.col0
 28
 82
 
-# TODO: WRONG
```
Comment (Dandandan) [inline datafusion/core/tests/sqllogictests/test_files/groupby.slt:1480]: Can remove link to issue

File: datafusion/core/tests/sqllogictests/test_files/groupby.slt [hunk 2]
```diff
@@ -1914,13 +1911,11 @@ SELECT ALL - ( 30 ) * + cor0.col1 AS col2 FROM tab2 AS cor0 GROUP BY cor0.col1
 -1770
 -1830
 
-# TODO: WRONG
```
Comment (Dandandan) [inline datafusion/core/tests/sqllogictests/test_files/groupby.slt:1917]: Can remove link to issue

Comment (mingmwang) [reply datafusion/core/tests/sqllogictests/test_files/groupby.slt:1917]: Sure

**Review by Dandandan — APPROVED** [review]

**Review by jackwener — APPROVED** [review]
Thank you @mingmwang

**Comment by alamb** [comment]
I think it is always awesome when bugs are fixed by deleting code

Screenshot 2023-04-25 at 10 45 35 AM

**Pull request merged**